\documentclass[11pt]{article}
\usepackage{bibentry}
\usepackage{url}
\begin{document}
\bibliographystyle{plain}
\nobibliography{manual.bib}

\title{ALF User Manual}
\author{Daniel A. Dalquen}
\maketitle

\section{Getting Started}
\subsection{Obtaining ALF}
ALF is available as a web service as well as a stand-alone version. Both are available at \url{http://www.cbrg.ethz.ch/alf}. The web service can be used to run simulations directly or to generate parameter files for the stand-alone version. I recommend using the web service only for small simulations. If you want to be notified when new releases are available, you can register with your email address, when you download the stand-alone version.

\subsection{Installing the stand-alone version}
When you have download and unpack the stand-alone version of ALF, the following
files and directories are created:

\begin{description}
\item[\texttt{install.sh}] \hfill \\
install script (instructions below)
\item[\texttt{bin/}] \hfill \\
       - binaries of the darwin engine for Mac OS X and Linux\hfill \\
	   - starter scripts for darwin and ALF
\item[\texttt{lib/}] \hfill \\
       - the darwin library\hfill \\
	   - the entire ALF source code is located in lib/simulator
\item[\texttt{params/}] \hfill \\
sample parameter sets to reproduce some of the results from the publication
\end{description}

ALF can be started directly from this directory (see section \ref{sec.run}), but it will probably be more convenient to install the software, because that will allow you to start ALF directly from any directory.

To install ALF on your system, run the script install.sh. By default, binary and script files will be copied to \texttt{/usr/local/bin} and the \texttt{lib} directory will be copied to \texttt{/usr/local/share/alfdarwin} (this will require super user privileges). If you wish to install ALF in a different location, pass the path to that location as parameter to \texttt{install.sh}. For example, if you want to install ALF in your home directory, use
\begin{quote}
\texttt{./install.sh /path/to/home}.
\end{quote}
This will create the directories \texttt{bin/} and \texttt{share/alfdarwin/} in your home directory, if they don't exist already.

\subsection{Running Simulations}
\label{sec.run}
If you did not install ALF, run it with the following command:
\begin{quote}
\begin{verbatim}
bin/alfsim [path to parameter file]
\end{verbatim}
\end{quote}
If you did install ALF and the script is on your PATH, then this reduces to
\begin{quote}
\begin{verbatim}
alfsim [path to parameter file]
\end{verbatim}
\end{quote}

When no paramater file is given, ALF will look for the file parameters.drw in
the current working directory. Section \ref{sec.params} lists and explains all available parameters.

\subsection{Citing ALF}
If you use ALF in your research, please cite:
\begin{quote}
\bibentry{Dalquen2011}
\end{quote}

\subsection{Contact information}
If you have a question or want to report a bug, please send
me an email to \url{ddalquen@inf.ethz.ch}. 


\section{Parameters}
\label{sec.params}
Parameters for a simulation are passed in a parameter file that uses the Darwin syntax (see \url{http://www.biorecipes.com/DarwinHelp} for details). For example,
\begin{quote}
\texttt{protStart := 200;}
\end{quote}
sets the parameter \texttt{protStart} to 200. Note that each assignment is terminated by a semicolon.

Some parameters expect a list (comma-separated values enclosed in square brackets, e.g. \texttt{[2.4, 133.8]}) or a set (comma-separated values in curly brackets, e.g. \texttt{\{0, 1, 2.5\}}).

Strings are enclosed in single quotes, e.g. \texttt{'a string'}.

Finally, comments start with the number sign (\verb|#|).


%#SetRand(12345): # use this with any number, if you want reproducable results
%
\subsection{Root Genome}
\begin{itemize}
\item{\texttt{realorganism}} \hfill \\
 A string specifying the path to the Darwin DB with the sequence data. Use the ALF web interface to convert Fasta files into the Darwin DB format.

Example:
\begin{quote}
\begin{verbatim}
realorganism := 'realseed/se_ECOLI_core.db';
\end{verbatim}
\end{quote}
\item{\texttt{areaPath}} \hfill \\
\[TODO\]
\item{\texttt{protStart}} \hfill \\
The number of sequences that the first organism should have (only if the root genome is generated).
\item{\texttt{gammaLengthDist}} \hfill \\
List of parameters for the length distribution of the generated sequences ($\sim\Gamma(k,\theta)$). 

Example:
\begin{quote}
\begin{verbatim}
gammaLengthDist := [2.4, 133.8];
\end{verbatim}
\end{quote}
If lengths of all sequences should be identical, set $k=\theta=1$ and specify the desired length with \texttt{minGeneLength}.

\item{\texttt{minGeneLength}} \hfill \\
Minimum length of a gene.
\end{itemize}

\subsection{Substitution Models}
\begin{itemize}
\item{\texttt{substModels}} \hfill \\
List of substitution models, where each model definition has the following format:
\begin{quote}
\begin{verbatim}
SubstitutionModel(name:string, parameters:list, 
                  frequencies:list, neutralDNA:boolean)
\end{verbatim}
\end{quote}
The number of arguments required depend on the model:
\begin{itemize}
\item models \texttt{CPAM}, \texttt{ECM}, \texttt{ECMu}, \texttt{GCB}, \texttt{JTT}, \texttt{WAG} and \texttt{LG} require just the name of the model (e.g. \texttt{SubstitutionModel('CPAM')}).
\item When using a custom matrix (\texttt{CustomC}, \texttt{CustomP}), pass the path to the matrix file as parameter.
\item For M-series models, pass also the codon frequencies.
\item Finally, for nucleotide models specify as fourth parameter whether non-sense mutations should be allowed.
\end{itemize}
Available models include:
\begin{description}
\item{nucleotide substitution} \hfill \\
\texttt{F84}, \texttt{GTR}, \texttt{HKY}, \texttt{TN93}
\item{codon substitution} \hfill \\
\texttt{CPAM}, \texttt{ECM}, \texttt{ECMu}, \texttt{M0}, \texttt{M2}, \texttt{M3}, \texttt{M8}, \texttt{CustomC}
\item{amino acid substitution} \hfill \\
\texttt{GCB}, \texttt{JTT}, \texttt{LG}, \texttt{WAG}, \texttt{CustomP}
\end{description}
Parameters are ordered as follows
\begin{description}
\item{custom empirical models} \hfill \\
\texttt{parameters[1]} should contain a path to a matrix in PAML format (lower triangular matrix of exchangabilitites, followed by a line with codon/aa frequencies).
\item{M-series models} \hfill \\
\texttt{parameters[1]}: $\kappa$ \hfill \\
\texttt{parameters[2]}: $\omega$ (single value or list) \hfill \\
\texttt{parameters[3]}: list of probabilities for $\omega$-class[es] \hfill \\
\texttt{parameters[4]}: for M8, parameter $p$ of beta distribution \hfill \\
\texttt{parameters[5]}: for M8, parameter $q$ of beta distribution \hfill \\
\item{nucleotide models} \hfill \\
\texttt{GTR}:
\begin{description}
\item{\texttt{parameters[1]}\ldots\texttt{parameters[6]}} \hfill \\
$a\ldots f$ \hfill \\
\end{description}
\texttt{HKY}:
\begin{description}
\item{\texttt{parameters[1]}:} $\alpha$ \hfill \\
\item{\texttt{parameters[2]}:} $\beta$ \hfill \\
\end{description}
\texttt{F84}: \hfill \\
\begin{description}
\item{\texttt{parameters[1]:}} $\kappa$ \hfill \\
\item{\texttt{parameters[2]:}} $\beta$ \hfill \\
\end{description}
\texttt{TN93}: \hfill \\
\begin{description}
\item{\texttt{parameters[1]:}} $\alpha_{1}$ \hfill \\
\item{\texttt{parameters[2]:}} $\alpha_{2}$ \hfill \\
\item{\texttt{parameters[3]:}} $\beta$ \hfill \\
\end{description}
\end{description}
 Example:
\begin{quote}
\begin{verbatim}
substModels := [SubstitutionModel('CPAM'),
                SubstitutionModel('TN93', [.3, .4, .7],
                                  [seq(0.25,4)], true)]:
\end{verbatim}
\end{quote}
\item{\texttt{modelAssignments}} \hfill \\
Parameter concerning selection of substitution model for each sequence of the root genome. Either supply a list of frequencies of models $1\ldots n$ defined above for random assignment or a list of assignments for each sequence of the root genome.

Example:
\begin{quote}
\begin{verbatim}
modelAssignments := [0.75, 0.25];
\end{verbatim}
\end{quote}
\item{\texttt{modelSwitchS}/\texttt{modelSwitchD}} \hfill \\
A matrix with probabilities for a switch from model $i$ to model $j$ after speciation or duplication, respectively.

Example:
\begin{quote}
\begin{verbatim}
modelSwitchS := [[0.8,0.2],[0,1]];
modelSwitchD := [[1,0],[0,1]]; # no switch
\end{verbatim}
\end{quote}
\item{\texttt{blocksize}} \hfill \\
When no substitution model is given (pure gap simulation), select block size for gaps
\end{itemize}

\subsection{Gap Model}
\begin{itemize}
\item{\texttt{indelModel}} \hfill \\
 \begin{description}
\item{\texttt{'ZIPF'}} \hfill \\

\item{\texttt{'NEGBIN'}} \hfill \\

\item{\texttt{'QG'}} \hfill \\

\item{\texttt{'EXP'}} \hfill \\

\item{\texttt{'CUSTOM'}} \hfill \\

\end{description}
\item{\texttt{indelVector}} \hfill \\
A list with probabilities defining a custom indel distribution. Should be \texttt{maxIndelLength} elements long.
\item{\texttt{Z\_c}} \hfill \\
Exponent of Zipfian distribution
\item{\texttt{QG\_c}} \hfill \\
c parameter for generalized Qian-Goldstein model in PAM (see Pang et al. BMC Bioinformatics, 6:235 2005)
\item{\texttt{QG\_t}} \hfill \\
c parameter for generalized Qian-Goldstein model in PAM (see Pang et al. BMC Bioinformatics, 6:235 2005)
\item{\texttt{NB\_r}} \hfill \\
r parameter for negative binomial model (must be positive integer)
\item{\texttt{NB\_q}} \hfill \\
q parameter for negative binomial model (0 < q < 1):
\item{\texttt{E\_p}} \hfill \\
proportion for geometric distribution with mean 1/E\_p
\item{\texttt{maxIndelLength}} \hfill \\
maximal length of indels

\item{\texttt{aaGainRate}} \hfill \\
rate that a gene gains 1 aa, p for single genes
\item{\texttt{aaLossRate}} \hfill \\
rate that a gene loses 1 aa, p for single genes
\end{itemize}

\subsection{GC Content Amelioration}
%targetFreqs := ['Random']: # creates random target frequencies for all leaf species (overrides frequencies supplied in substitution model above)
%# If you want to have specific target frequencies per species and substitution model, give an array with the following structure:
%# targetFeqs := [freqs_for_subst_model_1, freqs_for_subst_model_2, ...] with
%# freqs_for_subst_model_i = [['species_name_1, [list_of_frequencies_1]], ['species_name_2, [list_of_frequencies_2]], ...]
%# example for a simulation with 4 species using a nucleotide model:
%# targetFreqs := [[['S1',[0.15, 0.35, 0.3, 0.2]],
%#                 ['S2',[0.2, 0.25, 0.3, 0.25]],
%#                 ['S3',[0.25, 0.2, 0.25, 0.3]],
%#                 ['S4',[0.35, 0.15, 0.2, 0.3]]]]:
%

\subsection{Tree Parameters}
%treeType := 'BDTree':       # BDTree, ToLSample, Custom
%scaleTree := false:          # scale tree to match Pam distance defined below (parameter mutRate)
%birthRate := 0.01:            # b parameter (for BDTree)
%deathRate := 0.001:           # d parameter (for BDTree)
%NSpecies := 15:            # number of species in the tree (for BD and ToL)
%ultrametric := false:     # for BDTree: should resulting tree be ultrametric
%
%treeFile := 'tree.drw': # path to tree file with tree in darwin or Newick format OR a darwin tree structure
%# unitIsPam := true:   # set to false if branch lengths are in substitutions per site, set rate parameters accordingly.
%###
%
%
%# probabilities and evolution parameters
%mutRate := 100;        # PAM distance from origin to recent species (for random trees)

\subsection{Rate Variation}
%amongGeneDistr := 'Gamma': # distribution of rates among genes. Use 'None' for no variation or 'Custom'
%                           # for using custom rates from file stored at aGPath.
%aGAlpha := 1:              # Shape parameter of among gene distribution. The mean is always moved to 1
%                           # in order to keep the mean branch length.
%aGPath := 'realseed/customRates.drw': # path to file with custom rates (see example file, only for custom root sequences)
%amongSiteDistr := 'None'; # distribution of rates (Gamma or Poisson)
%                             # 'Poisson' generates a random number of domains (at most areas) per
%                             # gene with rates drawn from a Poisson distribution around the mean
%                             # rate (given by mutRate). motifFreq defines the fraction of domains
%                             # with mutation rate 0 (motif).
%                             # 'Gamma' uses gamma rates. The number of bins is defined by the areas
%                             # parameter. Additionally, motifs occur with frequency motifFreq.
%                             # If user-defined rates are given, this option is ignored.
%                             # For M-series models this option is ignored and replaced with the classes of the model
%areas := 5;                  # maximal number of areas with different mutRate within a gene
%motifFreq := 0.01:           # rate for of invariable sites
%alphaG := 1;                 # alpha parameter of gamma function

\subsection{Gene Duplication and Loss}
%geneDuplRate := 0.0005;    # rate that gene duplication occurs
%  transDupl := 0.5;     # ratio of tranlocation after duplication
%  numberDupl := 5;     # maximal number of genes involved in one duplication
%  fissionDupl := 0;   # rate of fission after duplication of a single gene
%  fusionDupl := 0;    # rate of fusion of two genes after duplication
%## duplicate evolves as pseudogene  (permanent rate change for duplicate)
%  P_pseudogene := 0: # probability of duplicate becoming a pseudogene
%  ratefac_pseudogene := 0.9: # rate change factor 
%## duplicate evolves under neofunctionalization (temporary rate change for duplicate)
%  P_neofunc := 0:          # probability of duplicate undergoing neofunctionalization
%  ratefac_neofunc := 1.5:    # rate change factor
%  life_neofunc := 10:        # life of increased rate (time to normalization of rate)
%## both copies evolving by subfunctionalization (temporary rate change for original and duplicate)
%  P_subfunc := 0:           # probability of both genes undergoing subfunctionalization
%  ratefac_subfunc := 1.2:     # rate change factor
%  life_subfunc := 10:         # life of rate change (time to normalization of rate)
%  
%geneLossRate := 0.0005;    # rate that gene loss occurs
%  numberLoss := 5:    # maximal number of genes involved in one duplication
%
\subsection{Lateral Gene Transfer}
%lgtRate := 0.0005;         # rate that single lateral gene transfer occurs
%  orthRep := 0;       # how much of the transfer will be orthologous
%                       # replacement (rest is novel gene acquisition)
%lgtGRate := 0.00005;	       # rate that lateral gene transfer of groups occurs
%  lgtGSize := 10;      # number of genes which are transferred in one go

\subsection{Genome Rearrangement}  
%invers := 0.00;	       # rate that gene inversion occurs
%  invSize := 1;        # number of genes which are inversed in one go
%transloc := 0.00;	       # rate that gene translocation occurs
%  transSize := 1;      # number of genes which are translocated in one go
%invtrans := 0.10;	       # rate that inversed translocation occurs

\subsection{Gene Fusion and Fission}
%fissionRate := 0;      # rate of gene fissions without prior duplication
%fusionRate := 0;       # rate of gene fusions occuring without prior duplication of fused genes
%  numberFusion := 3:

\subsection{Output}
The following parameters define what output is generated and where it is stored. ALF will always generate a species tree, that reflects the ancestry of the simulated species, and the set of genomes of the species at the leaves of that tree.

\begin{itemize}
\item{\texttt{simOutput}} \hfill \\
A set of output files and formats. If this variable is not defined, all outputs are generated. Possible values include:
\begin{description}
\item{\texttt{'GeneTrees'}}\hfill \\
all gene trees
\item{\texttt{'Ancestral'}}\hfill \\
output ancestral genomes
\item{\texttt{'MSA'}}\hfill \\
MSAs of all related sequences
\item{\texttt{'VP'}}\hfill \\
pairwise evolutionary relationships (ortho/para/xenologs)
\item{\texttt{'DarwinTree'}}\hfill \\
output trees in Darwin format
\item{\texttt{'Newick'}}\hfill \\
output trees in Newick format (default)
\item{\texttt{'DarwinDB'}}\hfill \\
output genomes as Darwin databases
\item{\texttt{'Fasta'}}\hfill \\
output genomes as Fasta files (default)
\end{description}

\noindent \textbf{Example} (creates species tree in Darwin format, MSAs of all gene families, and Darwin databases for all ancestral and leaf species):
\begin{quote}
\begin{verbatim}
simOutput := {'DarwinTree', 'MSA', 'DarwinDB', 'Ancestral'};
\end{verbatim}
\end{quote}

\item{\texttt{wdir}} \hfill \\
A string specifying the working directory, i.e. the directory where simulation results are stored. current working directory by default, can also be set as argument of alfsim

\item{\texttt{mname}} \hfill \\
A string specifying the name of the simulation. ALF will create a directory of that name in the working directory containing all result files.
\end{itemize}
\end{document}
